Add: Mobile responsive Funds page

willm30 opened the pull request:
## Description

This PR adresses the mobile responsive design for the Funds page. There is no design for this one, so I've freestyled a bit (e.g. with the `ColonyFundingMenu` list items). 

**Changes** 🏗

* Adapting `ColonyFunding` and related components for mobile
* Refactor `ColonyFundingMenu` item list

## Screenshots

**_Funds page on mobile_**
![funds-mob](https://user-images.githubusercontent.com/64402732/176225718-8c641c28-e6db-4e2e-856a-9133b2e3d77d.png)

**_Info popover_**
![fund-info-popover](https://user-images.githubusercontent.com/64402732/176225696-38d7e94b-0cd9-4238-be37-fbd5c2168ca8.png)

Resolves #3494

arrenv reviewed (changes requested):
Mostly good with the freestyle, I played around with it in browser and removed the `box-shadows`, what are your thoughts on this screenshot?

![cd51be2d-b44d-4586-8e71-e8bddb06cb05](https://user-images.githubusercontent.com/33682027/176185047-b86aa87c-f641-42e9-8db2-a166970e7c46.png)
.

willm30 commented on src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx:
```diff
@@ -120,54 +121,46 @@ const ColonyFundingMenu = ({
-            !canUserMintNativeToken ||
-            !isSupportedColonyVersion ||
-            !isNetworkAllowed ||
-            !hasRegisteredProfile ||
-            !isDeploymentFinished ||
-            mustUpgradeOneTx
-          }
-        />
-      </li>
-      <li>
-        <Button
-          text={MSG.navItemManageTokens}
-          appearance={{ theme: 'blue' }}
-          onClick={handleEditTokens}
-          disabled={
-            !canEdit ||
-            !isSupportedColonyVersion ||
-            !isNetworkAllowed ||
-            !hasRegisteredProfile ||
-            !isDeploymentFinished ||
-            mustUpgradeOneTx
-          }
-          data-test="manageTokens"
-        />
-      </li>
+      <FundingItem
+        text={MSG.navItemMoveTokens}
+        handleClick={handleMoveTokens}
+        disabled={!canMoveTokens}
+      />
+      <FundingItem
+        text={MSG.navItemMintNewTokens}
+        handleClick={handleMintTokens}
+        disabled={!canUserMintNativeToken}
+      />
+      <FundingItem
+        text={MSG.navItemManageTokens}
+        handleClick={handleEditTokens}
+        disabled={!canEdit}
+      />
```
I switched to `master` branch to check this behaviour, and on master I see the same as your screenshot when I am signed in to a wallet that is a member of the colony, but not the colony's creator. When I am signed in as the creator of the colony, all three links are active. 

This is with the governance extension installed and no permissions set for the non-creator member (and no reputation in the colony). 

I presume that with the governance extension installed, all colony members (not just creator) should be able to create a motion to move funds and manage tokens? If so, it seems like I should open a separate issue to address this.

arrenv replied:
Yeah, it is a separate issue, I wasn't really clear about it my comment though, sorry.

willm30 replied:
Ok will keep track here: https://github.com/JoinColony/colonyDapp/issues/3550

willm30 commented:
@arrenv Looks good! I've updated and think I've captured all your changes. The text at the bottom wraps for me at small sizes (under 400px) but apart from that I think I've updated the code to mirror your screenshot .

rdig commented on src/modules/dashboard/components/ColonyFunding/ColonyFunding.css:
```diff
@@ -49,3 +50,134 @@
+  .main div[class*="InfoPopover_section"] {
+    display: flex;
+    flex-direction: column;
+    background: white;
+    gap: 10px;
+  }
+
+  .main div[class*="InfoPopover_displayName"] {
+    display: flex;
+    align-items: center;
+    gap: 5px;
+  }
+
+  .main div[class*="InfoPopover_displayName"] figure {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .main div[class*="Avatar_image"] {
+    position: static;
+  }
+
+  .main div[class*="CopyableAddress_main"] {
+    align-items: flex-start;
+    gap: 30px;
+  }
+
+  .main div[class*="CopyableAddress_addressContainer"] {
+    overflow-wrap: anywhere;
+  }
+
+  .main span[class*="CopyableAddress_copyButton"] {
+    padding-right: 10px;
+    line-height: 0px;
+  }
+
+  .main span[class*="CopyableAddress_copyButton"] button {
+    line-height: 10px;
+  }
```
Yeah, this won't work.

I left comments regarding this in the other PR

willm30 replied:
I've changed them to track the current state of the dom. In the case of the `popover` I used [role="tooltip"] on the basis that this attribute is likely to be stable (& doesn't rely on postcss) and because the `popover`'s presence in the dom is conditional on a user's click, which makes it harder to select without affecting the element's around it.

rdig commented on src/modules/dashboard/components/ColonyFunding/ColonyFunding.css:
```diff
@@ -49,3 +50,134 @@
+    padding: 0;
+  }
+
+  .banner {
+    margin: 0px;
+  }
```
When there are multiple token entries on the page, making the page scroll, this element has too little bottom margin.

I think we should increase it.

![Screenshot from 2022-06-29 15-19-28](https://user-images.githubusercontent.com/1193222/176435100-f86f6267-6264-47e5-b619-212dba67c2c5.png)
![Screenshot from 2022-06-29 15-19-31](https://user-images.githubusercontent.com/1193222/176435107-6574bfa2-868a-4a40-aa5f-d0fbc3cf0cda.png)

ArmandoGraterol commented on src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx:
```diff
@@ -120,54 +121,46 @@ const ColonyFundingMenu = ({
+        text={MSG.navItemManageTokens}
+        handleClick={handleEditTokens}
+        disabled={!canEdit}
+      />
```
Nice optimization :ok_hand:

ArmandoGraterol reviewed (approved):
Nice freestyle yo!

chinins reviewed (approved):
All good!

willm30 commented:
@arrenv @rdig Would I be able to get a re-review (pretty please 🙂)?
